packages/react/src/dialog/Dialog.tsx

```diff
@@ -0,0 +1,50 @@
+import * as RadixDialog from "@radix-ui/react-dialog";
+import { type ComponentPropsWithRef, type ReactNode, forwardRef } from "react";
+
+import type { ExtendProps } from "../utils";
+
+import { Box } from "../box";
+import { Flex } from "../flex";
+import { CloseIcon } from "./CloseIcon";
+import * as styles from "./Dialog.css";
+
+type DialogProps = ExtendProps<
+  ComponentPropsWithRef<typeof Box>,
+  ComponentPropsWithRef<typeof RadixDialog.Root>,
+  {
+    children: ReactNode;
+    hideCloseIcon?: boolean;
```

Comment:
let's invert this - change this `withCloseButton` and by default it'll be `false`

so we won't show the close button by default - but users can opt-in to that